apps/judicial-system/backend/src/app/modules/subpoena/subpoena.service.ts

```diff
@@ -320,4 +334,39 @@ export class SubpoenaService {
       return { delivered: false }
     }
   }
+
+  async getSubpoena(subpoena: Subpoena, user?: TUser): Promise<Subpoena> {
+    if (!subpoena.subpoenaId) {
+      // The subpoena has not been delivered to the police
+      return subpoena
+    }
+
+    if (subpoena.serviceStatus) {
```

Comment:
Please consider combining these two if statements into one for better readability

Reply:
I prefer to keep them separate as the two reason are truly different. The reasons are not obvious from the conditions, so I added comments, which  I think are more readable if kept separate.